### PR TITLE
Decrease cosmos MaxParallel from 8 to 4

### PR DESCRIPTION
**Pull request by mikeharder**

### CODE REVIEW

**Comment by kushagraThapar** [comment]
@mikeharder  - please do not merge this PR until we reach to a different solution of handling the agents issue. 
+ @moderakh

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
Sorry to block this PR, no offense. 
Once we reach to a conclusion, I will unblock it.

**Comment by mikeharder** [comment]
Closing until agreement is reached on next steps

**Pull request closed without merging**
